components/processor/gcomapiprocessor/internal/gcom/client/client.go

```diff
@@ -0,0 +1,154 @@
+package client
+
+import (
+	"context"
+	"flag"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/opentelemetry-collector-components/processor/gcomapiprocessor/internal/gcom/common"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	grafanaComReqs = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "grafanacom_api_request_duration_seconds",
+			Help:    "A histogram of request latencies by status_code and path.",
+			Buckets: prometheus.DefBuckets,
+		},
+		[]string{"code", "path"},
+	)
+)
+
+// Config contains the configuration required to
+// create a grafana.com Client
+type Config struct {
+	Key      string
+	Endpoint string
+	Timeout  time.Duration
+	Client   *http.Client
+
+	// Mock allows for a mock client to be substituted
+	// and returned by the New function
+	Mock Client
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.Key, "grafanacloud.client.key", "", "API key for accessing grafana.com")
+	f.StringVar(&cfg.Endpoint, "grafanacloud.client.endpoint", "https://www.grafana.com/api", "URL for grafana.com api")
+	f.DurationVar(&cfg.Timeout, "grafanacloud.client.timeout", 1*time.Minute, "Timeout for requests to grafana.com api")
+}
+
+// Client is used to interact with the GrafanaCloud api
+type Client interface {
+	// CheckAPIKey checks a Grafana Cloud API key and returns its parsed representation.
+	CheckAPIKey(ctx context.Context, key string) (*APIKey, error)
+
+	// ListInstances returns a list of Grafana Cloud instances from API.
+	ListInstances(ctx context.Context, options InstanceRequestOptions) ([]Instance, error)
+
+	// ListInstancesWithPagination returns a list of Grafana Cloud instances from API,
+	// but it requests them one page at a time to avoid a timeout.
+	ListInstancesWithPagination(ctx context.Context, options InstanceRequestOptions) ([]Instance, error)
+
+	// GetInstance returns a single Grafana Cloud instance with the ID provided in the options.
+	// If the ID is not provided, an error is returned.
+	GetInstance(ctx context.Context, options InstanceRequestOptions) (Instance, error)
+
+	// ListOrgs returns a list of Grafana Cloud organizations from API.
+	ListOrgs(ctx context.Context, options *OrgRequestOptions) ([]Org, error)
+}
+
+type client struct {
+	key      string
+	endpoint *url.URL
+	client   *http.Client
+	logger   log.Logger
+}
+
+// New creates a grafana.com Client
+func New(cfg Config, name string, logger log.Logger) (Client, error) {
```

Comment:
I just realized that those are the two packages you said you just copied from the enterprise package. If we are going to maintain them, should we accomodate them to the collector standards? Or do we want to threat those two packages as libraries? In that case, it would probably make sense to split them into different go modules perhaps.

Reply:
I'll follow up with a PR against backend-enterprise to convert those packages to modules. If it wouldn't work for some reason, let's improve the code and strip all unnecessary parts.